### PR TITLE
Fix CI: remove tfp converter and fix mypy

### PR DESCRIPTION
**Pull request by roshnaeem**
## Description
Replaced the current `mypy.ini` file with `mypy.ini` downloaded from the artifacts.

EDIT (Oriol): Fixes CI that was broken due to outdated tfp code. Closes https://github.com/arviz-devs/arviz/issues/1474

### CODE REVIEW

**Comment by OriolAbril** [comment]
Removed the tfp converter to fix CI (which also allowed reactivating tests for pyjags that had to be skipped when pinning down tfp). I don't think much of the tfp related code was useful now, so it's probably better to take converters in tfp docs or the one in #1890 to substitute the current one completely

**Comment by canyon289** [comment]
Ill approve this given the TF version is old and its holding up the entire ArviZ project. If we need to add it back in we have this pr as reference and its easy enough to revert

**Review by canyon289 — APPROVED** [review]

File: arviz/plots/dotplot.py [hunk 1]
```diff
@@ -222,9 +222,7 @@ def layout_stacks(stack_locs, stack_counts, binwidth, stackratio, rotated):
     binradius = binwidth / 2
 
     x = np.repeat(stack_locs, stack_counts)
-    y = np.hstack(
-        np.array([dotheight * np.arange(count) + binradius for count in stack_counts], dtype=object)
-    )
+    y = np.hstack([dotheight * np.arange(count) + binradius for count in stack_counts])
```
Comment (canyon289) [inline arviz/plots/dotplot.py:225]: What happened here? Just curious

**Comment by canyon289** [comment]
This can be fixed on followup PRs but we should remove these other tensorflow references as well
https://github.com/arviz-devs/arviz/blob/main/.azure-pipelines/azure-pipelines-external.yml#L31

File: arviz/plots/dotplot.py (hunk 1 above)
Comment (OriolAbril) [reply arviz/plots/dotplot.py:225]: numpy added type hints, and array was not on the accepted list of input types

**Pull request merged**